core/src/it/units/sdm/project/game/GameStatusHandler.java

```diff
@@ -0,0 +1,51 @@
+package it.units.sdm.project.game;
+
+import it.units.sdm.project.board.FreedomBoardHelper;
+import it.units.sdm.project.board.Position;
+import it.units.sdm.project.enums.GameStatus;
+import it.units.sdm.project.game.gui.FreedomGame;
+import org.jetbrains.annotations.NotNull;
+
+public class GameStatusHandler {
+
+    @NotNull
+    private final FreedomGame game;
+    @NotNull
+    private GameStatus status = GameStatus.FREEDOM;
+
+    public GameStatusHandler(@NotNull FreedomGame game) {
+        this.game = game;
+    }
+
+    @NotNull
+    public GameStatus getStatus() {
+        return status;
+    }
+
+    public void proceedToNextState() {
+        switch (status) {
+            case LAST_MOVE:
+                status = GameStatus.PLAY_LAST_MOVE;
+                break;
+            case PLAY_LAST_MOVE:
+                status = GameStatus.GAME_OVER;
+                break;
+            case GAME_OVER:
+                status = GameStatus.FREEDOM;
+                break;
+            case FREEDOM:
+            case NO_FREEDOM:
+                Position lastChosenPosition = game.getPlayersMovesHistory().getLast().getPosition();
+                if (FreedomBoardHelper.areAdjacentCellsOccupied(game.getBoard(), lastChosenPosition)) {
+                    status = GameStatus.FREEDOM;
+                } else if (FreedomBoardHelper.getNumberOfFreeCells(game.getBoard()) == 1) {
+                    status = GameStatus.LAST_MOVE;
+                } else {
+                    status = GameStatus.NO_FREEDOM;
+                }
+                break;
+            default:
```

Comment:
It can be either removed or replaced with the upper cases (`FREEDOM` and `NO_FREEDOM`)

Reply:
? You mean the `default` case?

Reply:
yep